src/node_quirc_decode.c

```diff
@@ -415,3 +421,49 @@ nq_load_jpeg(struct quirc *q, const uint8_t *img, size_t img_len)
 	jpeg_destroy_decompress(&dinfo);
 	return -1;
 }
+
+static int
+nq_load_raw(struct quirc *q, const uint8_t *img, size_t img_len, size_t width, size_t height)
+{
+	if (quirc_resize(q, width, height) < 0)
+		goto fail;
+
+	uint8_t *image = quirc_begin(q, NULL, NULL);
+
+	int channels = img_len / width / height;
+
+	if (channels == 1)
+	{
+		memcpy(image, img, img_len);
+	}
+	else if (channels == 3 || channels == 4)
+	{
+		size_t dst_offset = width * height - 1;
+		size_t src_offset = dst_offset * channels;
+		do
```

Comment:
going backward make this loop awkward, it could be a simple `for` right?

Reply:
Yeah, I was being a little too clever trying to avoid a potential perf issue. I'll simplify it.